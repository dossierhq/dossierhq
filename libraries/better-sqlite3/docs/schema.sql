CREATE TABLE subjects (
    id INTEGER PRIMARY KEY,
    uuid TEXT NOT NULL,
    created_at TEXT NOT NULL,
    CONSTRAINT subjects_uuid UNIQUE (uuid)
  ) STRICT;
CREATE TABLE principals (
    id INTEGER PRIMARY KEY,
    provider TEXT NOT NULL,
    identifier TEXT NOT NULL,
    subjects_id INTEGER NOT NULL,
    CONSTRAINT principals_pkey UNIQUE (provider, identifier),
    FOREIGN KEY (subjects_id) REFERENCES subjects(id) ON DELETE CASCADE
  ) STRICT;
CREATE TABLE schema_versions (
    id INTEGER PRIMARY KEY,
    specification TEXT NOT NULL
  ) STRICT;
CREATE TABLE sequences (
    name TEXT NOT NULL UNIQUE,
    value INTEGER DEFAULT 0
  ) STRICT;
CREATE TABLE entities (
    id INTEGER PRIMARY KEY,
    uuid TEXT NOT NULL,
    name TEXT NOT NULL,
    type TEXT NOT NULL,
    auth_key TEXT NOT NULL,
    resolved_auth_key TEXT NOT NULL,
    status TEXT NOT NULL,
    never_published INTEGER NOT NULL DEFAULT TRUE,
    created_at TEXT NOT NULL,
    updated_at TEXT NOT NULL,
    updated_seq INTEGER NOT NULL,
    latest_entity_versions_id INTEGER,
    published_entity_versions_id INTEGER, valid INTEGER NOT NULL DEFAULT TRUE, revalidate INTEGER NOT NULL DEFAULT FALSE,
    CONSTRAINT entities_uuid UNIQUE (uuid),
    CONSTRAINT entities_name UNIQUE (name),
    CONSTRAINT entities_updated_seq UNIQUE (updated_seq),
    FOREIGN KEY (latest_entity_versions_id) REFERENCES entity_versions(id),
    FOREIGN KEY (published_entity_versions_id) REFERENCES entity_versions(id)
  ) STRICT;
CREATE VIRTUAL TABLE entities_latest_fts USING fts5 (
    content
  )
/* entities_latest_fts(content) */;
CREATE TABLE IF NOT EXISTS 'entities_latest_fts_data'(id INTEGER PRIMARY KEY, block BLOB);
CREATE TABLE IF NOT EXISTS 'entities_latest_fts_idx'(segid, term, pgno, PRIMARY KEY(segid, term)) WITHOUT ROWID;
CREATE TABLE IF NOT EXISTS 'entities_latest_fts_content'(id INTEGER PRIMARY KEY, c0);
CREATE TABLE IF NOT EXISTS 'entities_latest_fts_docsize'(id INTEGER PRIMARY KEY, sz BLOB);
CREATE TABLE IF NOT EXISTS 'entities_latest_fts_config'(k PRIMARY KEY, v) WITHOUT ROWID;
CREATE VIRTUAL TABLE entities_published_fts USING fts5 (
    content
  )
/* entities_published_fts(content) */;
CREATE TABLE IF NOT EXISTS 'entities_published_fts_data'(id INTEGER PRIMARY KEY, block BLOB);
CREATE TABLE IF NOT EXISTS 'entities_published_fts_idx'(segid, term, pgno, PRIMARY KEY(segid, term)) WITHOUT ROWID;
CREATE TABLE IF NOT EXISTS 'entities_published_fts_content'(id INTEGER PRIMARY KEY, c0);
CREATE TABLE IF NOT EXISTS 'entities_published_fts_docsize'(id INTEGER PRIMARY KEY, sz BLOB);
CREATE TABLE IF NOT EXISTS 'entities_published_fts_config'(k PRIMARY KEY, v) WITHOUT ROWID;
CREATE TABLE entity_versions (
    id INTEGER PRIMARY KEY,
    entities_id INTEGER NOT NULL,
    version INTEGER NOT NULL,
    created_at TEXT NOT NULL,
    created_by INTEGER NOT NULL,
    fields TEXT NOT NULL,
    FOREIGN KEY (entities_id) REFERENCES entities(id) ON DELETE CASCADE,
    FOREIGN KEY (created_by) REFERENCES subjects(id)
  ) STRICT;
CREATE TABLE entity_publishing_events (
    id INTEGER PRIMARY KEY,
    entities_id INTEGER NOT NULL,
    entity_versions_id INTEGER,
    published_by INTEGER NOT NULL,
    published_at TEXT NOT NULL,
    kind TEXT NOT NULL,
    FOREIGN KEY (entities_id) REFERENCES entities(id) ON DELETE CASCADE,
    FOREIGN KEY (entity_versions_id) REFERENCES entity_versions(id) ON DELETE CASCADE,
    FOREIGN KEY (published_by) REFERENCES subjects(id)
  ) STRICT;
CREATE TABLE advisory_locks (
    id INTEGER PRIMARY KEY,
    name TEXT NOT NULL,
    handle INTEGER NOT NULL,
    acquired_at TEXT NOT NULL,
    renewed_at TEXT NOT NULL,
    expires_at INTEGER NOT NULL,
    lease_duration INTEGER NOT NULL,
    CONSTRAINT advisory_locks_name UNIQUE (name)
  ) STRICT;
CREATE TABLE entity_published_references (
    id INTEGER PRIMARY KEY,
    from_entities_id INTEGER NOT NULL,
    to_entities_id INTEGER NOT NULL,
    FOREIGN KEY (from_entities_id) REFERENCES entities(id) ON DELETE CASCADE,
    FOREIGN KEY (to_entities_id) REFERENCES entities(id) ON DELETE CASCADE
  ) STRICT;
CREATE TABLE entity_latest_references (
    id INTEGER PRIMARY KEY,
    from_entities_id INTEGER NOT NULL,
    to_entities_id INTEGER NOT NULL,
    FOREIGN KEY (from_entities_id) REFERENCES entities(id) ON DELETE CASCADE,
    FOREIGN KEY (to_entities_id) REFERENCES entities(id) ON DELETE CASCADE
  ) STRICT;
CREATE TABLE entity_published_locations (
    id INTEGER PRIMARY KEY,
    entities_id INTEGER NOT NULL,
    lat REAL NOT NULL,
    lng REAL NOT NULL,
    FOREIGN KEY (entities_id) REFERENCES entities(id) ON DELETE CASCADE
  ) STRICT;
CREATE TABLE entity_latest_locations (
    id INTEGER PRIMARY KEY,
    entities_id INTEGER NOT NULL,
    lat REAL NOT NULL,
    lng REAL NOT NULL,
    FOREIGN KEY (entities_id) REFERENCES entities(id) ON DELETE CASCADE
  ) STRICT;
CREATE TABLE unique_index_values (
    id INTEGER PRIMARY KEY,
    entities_id INTEGER NOT NULL,
    index_name TEXT NOT NULL,
    value TEXT NOT NULL,
    latest INTEGER NOT NULL DEFAULT FALSE,
    published INTEGER NOT NULL DEFAULT FALSE,
    FOREIGN KEY (entities_id) REFERENCES entities(id) ON DELETE CASCADE,
    CONSTRAINT unique_index_values_index_value UNIQUE (index_name, value)
  ) STRICT;
CREATE INDEX unique_index_values_entities_id ON unique_index_values(entities_id);
CREATE INDEX entity_versions_entities_id ON entity_versions(entities_id);
CREATE INDEX entity_published_references_from_entities_id ON entity_published_references(from_entities_id);
CREATE INDEX entity_published_locations_entities_id ON entity_published_locations(entities_id);
CREATE INDEX entity_latest_references_from_entities_id ON entity_latest_references(from_entities_id);
CREATE INDEX entity_latest_locations_entities_id ON entity_latest_locations(entities_id);
CREATE INDEX entities_resolved_auth_key ON entities(resolved_auth_key);
CREATE INDEX entity_publishing_events_entities_id ON entity_publishing_events(entities_id);
CREATE INDEX entities_resolved_auth_key_name ON entities(resolved_auth_key, name);
CREATE INDEX entities_resolved_auth_key_updated_seq ON entities(resolved_auth_key, updated_seq);
CREATE INDEX entities_resolved_auth_uuid ON entities(resolved_auth_key, uuid);
