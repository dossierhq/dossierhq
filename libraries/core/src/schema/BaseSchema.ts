import type {
  ComponentTypeSpecification,
  FieldSpecification,
  PublishedSchemaSpecification,
  SchemaIndexSpecification,
  SchemaPatternSpecification,
  SchemaSpecification,
  SchemaSpecificationWithMigrations,
} from './SchemaSpecification.js';

export class BaseSchema<
  T extends SchemaSpecification | SchemaSpecificationWithMigrations | PublishedSchemaSpecification,
> {
  readonly spec: T;
  private cachedPatternRegExps: Record<string, RegExp> = {};

  constructor(spec: T) {
    this.spec = spec;
  }

  getEntityTypeCount(): number {
    return this.spec.entityTypes.length;
  }

  getEntityTypeSpecification(type: string): T['entityTypes'][number] | null {
    return this.spec.entityTypes.find((it) => it.name === type) ?? null;
  }

  getEntityFieldSpecification(
    entitySpec: T['entityTypes'][number],
    fieldName: string,
  ): T['entityTypes'][number]['fields'][number] | null {
    return entitySpec.fields.find((it) => it.name === fieldName) ?? null;
  }

  getComponentTypeCount(): number {
    return this.spec.componentTypes.length;
  }

  getComponentTypeSpecification(type: string): T['componentTypes'][number] | null {
    return this.spec.componentTypes.find((it) => it.name === type) ?? null;
  }

  getComponentFieldSpecification(
    componentSpec: ComponentTypeSpecification,
    fieldName: string,
  ): FieldSpecification | null {
    return componentSpec.fields.find((it) => it.name === fieldName) ?? null;
  }

  getPattern(name: string): SchemaPatternSpecification | null {
    return this.spec.patterns.find((it) => it.name === name) ?? null;
  }

  getPatternRegExp(name: string): RegExp | null {
    let regexp = this.cachedPatternRegExps[name];
    if (regexp) return regexp;

    const pattern = this.getPattern(name);
    if (!pattern) return null;

    regexp = new RegExp(pattern.pattern);
    this.cachedPatternRegExps[name] = regexp;
    return regexp;
  }

  getIndex(name: string): SchemaIndexSpecification | null {
    return this.spec.indexes.find((it) => it.name === name) ?? null;
  }
}
