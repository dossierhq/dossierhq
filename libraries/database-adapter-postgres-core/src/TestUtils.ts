import type { TransactionContext } from '@jonasb/datadata-server';
import { ServerTestUtils } from '@jonasb/datadata-server';
import type { PostgresDatabaseAdapter } from '.';
import { createPostgresDatabaseAdapterAdapter } from '.';
import type { UniqueConstraints } from './DatabaseSchema';

type QueryFn = PostgresDatabaseAdapter['query'];

interface MockedPostgresDatabaseAdapter extends PostgresDatabaseAdapter {
  query: jest.MockedFunction<QueryFn>;
}

export class MockUniqueViolationOfConstraintError extends Error {
  readonly uniqueConstraint: UniqueConstraints;

  constructor(message: string, uniqueConstraint: UniqueConstraints) {
    super(message);
    this.name = 'MockUniqueViolationOfConstraintError';
    this.uniqueConstraint = uniqueConstraint;
  }
}

export function createMockContext(adapter: PostgresDatabaseAdapter): TransactionContext {
  const databaseAdapter = createPostgresDatabaseAdapterAdapter(adapter);
  return ServerTestUtils.createDummyContext(databaseAdapter);
}

export function createMockAdapter(): MockedPostgresDatabaseAdapter {
  const query: jest.MockedFunction<QueryFn> = jest.fn();
  return {
    disconnect: jest.fn(),
    isUniqueViolationOfConstraint: (error, constraintName) =>
      error instanceof MockUniqueViolationOfConstraintError &&
      error.uniqueConstraint === constraintName,
    query,
    createTransaction: async () => ({ _type: 'Transaction', release: jest.fn() }),
  };
}

export function getQueryCalls(adapter: MockedPostgresDatabaseAdapter): [string, ...unknown[]][] {
  return adapter.query.mock.calls.map((call) => {
    const [_transaction, query, values] = call;
    return [query, ...(values ?? [])];
  });
}
