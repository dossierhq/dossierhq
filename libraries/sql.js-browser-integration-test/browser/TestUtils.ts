import { randomUUID } from 'node:crypto';
import { NoOpLogger, type ErrorType, type PromiseResult } from '@dossierhq/core';
import type { TestSuite } from '@dossierhq/integration-test';
import { createSqlJsAdapter, type SqlJsDatabaseAdapter } from '@dossierhq/sql.js';
import { test } from '@playwright/test';
import * as SqlJs from 'sql.js';

export async function createSqlJsTestAdapter(): PromiseResult<
  SqlJsDatabaseAdapter,
  typeof ErrorType.BadRequest | typeof ErrorType.Generic
> {
  const SQL = await SqlJs.default();
  const db = new SQL.Database();
  return await createSqlJsAdapter({ logger: NoOpLogger }, db, {
    migrate: true,
    fts: { version: 'fts4' },
    journalMode: 'memory',
  });
}

export function registerTestSuite(testSuiteName: string, testSuite: TestSuite): void {
  polyfillCrypto();
  test.describe(testSuiteName, () => {
    for (const [testName, testFunction] of Object.entries(testSuite)) {
      test(testName, testFunction);
    }
  });
}

function polyfillCrypto() {
  // This package is meant to be run in a browser, polyfill crypto for Node
  if (!globalThis.crypto) {
    globalThis.crypto = {
      randomUUID,
    } as Crypto;
  }
}

export function assertIsDefined<T>(val: T): asserts val is NonNullable<T> {
  if (val === undefined || val === null) {
    throw new Error(
      `Expected 'val' to be defined, but received ${val === undefined ? 'undefined' : 'null'}`,
    );
  }
}
