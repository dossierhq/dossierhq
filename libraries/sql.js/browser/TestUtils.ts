import type { ErrorType, PromiseResult } from '@dossierhq/core';
import { NoOpLogger } from '@dossierhq/core';
import type { TestSuite } from '@dossierhq/integration-test';
import { test } from '@playwright/test';
import { randomUUID } from 'node:crypto';
import * as SqlJs from 'sql.js';
import type { SqlJsDatabaseAdapter } from '../src/SqlJsAdapter.js';
import { createSqlJsAdapter } from '../src/SqlJsAdapter.js';

export async function createSqlJsTestAdapter(): PromiseResult<
  SqlJsDatabaseAdapter,
  typeof ErrorType.BadRequest | typeof ErrorType.Generic
> {
  const SQL = await SqlJs.default();
  const db = new SQL.Database();
  return await createSqlJsAdapter({ logger: NoOpLogger }, db, {
    migrate: true,
    fts: { version: 'fts4' },
    journalMode: 'memory',
  });
}

export function registerTestSuite(testSuiteName: string, testSuite: TestSuite): void {
  polyfillCrypto();
  test.describe(testSuiteName, () => {
    for (const [testName, testFunction] of Object.entries(testSuite)) {
      test(testName, testFunction);
    }
  });
}

function polyfillCrypto() {
  // This package is meant to be run in a browser, polyfill crypto for Node
  if (!globalThis.crypto) {
    globalThis.crypto = {
      randomUUID,
    } as Crypto;
  }
}
