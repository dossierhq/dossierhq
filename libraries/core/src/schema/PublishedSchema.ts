import { type PublishedSchemaSpecification } from './SchemaSpecification.js';
import { BaseSchema } from './BaseSchema.js';

export class PublishedSchema extends BaseSchema<PublishedSchemaSpecification> {
  constructor(spec: PublishedSchemaSpecification) {
    super(spec);
  }
}
