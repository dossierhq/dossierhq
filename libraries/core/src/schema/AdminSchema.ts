import { ok, type ErrorType, type Result } from '../ErrorResult.js';
import { BaseSchema } from './BaseSchema.js';
import { PublishedSchema } from './PublishedSchema.js';
import type {
  AdminSchemaSpecification,
  AdminSchemaSpecificationUpdate,
  AdminSchemaSpecificationWithMigrations,
} from './SchemaSpecification.js';
import { schemaAdminToPublished } from './schemaAdminToPublished.js';
import { schemaUpdate } from './schemaUpdate.js';
import { schemaValidateAdmin } from './schemaValidateAdmin.js';

export class AdminSchema<
  TSpec extends
    | AdminSchemaSpecification
    | AdminSchemaSpecificationWithMigrations = AdminSchemaSpecification,
> extends BaseSchema<TSpec> {
  private cachedPublishedSchema: PublishedSchema | null = null;

  static createAndValidate(
    update: AdminSchemaSpecificationUpdate,
  ): Result<AdminSchema, typeof ErrorType.BadRequest> {
    const result = AdminSchemaWithMigrations.createAndValidate(update);
    if (!result.isOk()) return result;

    const { migrations, ...specWithoutMigrations } = result.value.spec;

    return ok(new AdminSchema(specWithoutMigrations));
  }

  constructor(spec: TSpec) {
    super(spec);
  }

  validate(): Result<void, typeof ErrorType.BadRequest> {
    return schemaValidateAdmin(this);
  }

  toPublishedSchema(): PublishedSchema {
    if (this.cachedPublishedSchema) {
      return this.cachedPublishedSchema;
    }

    const publishedSpec = schemaAdminToPublished(this);

    this.cachedPublishedSchema = new PublishedSchema(publishedSpec);
    return this.cachedPublishedSchema;
  }
}

export class AdminSchemaWithMigrations extends AdminSchema<AdminSchemaSpecificationWithMigrations> {
  static override createAndValidate(
    update: AdminSchemaSpecificationUpdate,
  ): Result<AdminSchemaWithMigrations, typeof ErrorType.BadRequest> {
    const emptySpec: AdminSchemaSpecificationWithMigrations = {
      schemaKind: 'admin',
      version: 0,
      entityTypes: [],
      valueTypes: [],
      patterns: [],
      indexes: [],
      migrations: [],
    };
    const empty = new AdminSchemaWithMigrations(emptySpec);
    return empty.updateAndValidate(update);
  }

  updateAndValidate(
    update: AdminSchemaSpecificationUpdate,
  ): Result<AdminSchemaWithMigrations, typeof ErrorType.BadRequest> {
    // Update
    const updatedResult = schemaUpdate(this.spec, update);
    if (updatedResult.isError()) return updatedResult;
    const updatedSpec = updatedResult.value;

    if (updatedSpec === this.spec) {
      return ok(this); // no change
    }

    // Validate
    const updatedSchema = new AdminSchemaWithMigrations(updatedSpec);
    const validateResult = updatedSchema.validate();
    if (validateResult.isError()) return validateResult;

    return ok(updatedSchema);
  }
}
