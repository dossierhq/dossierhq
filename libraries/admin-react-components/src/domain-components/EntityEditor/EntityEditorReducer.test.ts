import { AdminEntityStatus } from '@jonasb/datadata-core';
import { Temporal } from '@js-temporal/polyfill';
import schema from '../../stories/StoryboardSchema';
import { bar1Id } from '../../test/EntityFixtures';
import { insecureTestUuidv4 } from '../../test/TestUtils';
import type { EntityEditorState } from './EntityEditorReducer';
import {
  AddEntityDraftAction,
  initializeEntityEditorState,
  reduceEntityEditorState,
  SetNameAction,
  UpdateEntityAction,
} from './EntityEditorReducer';

function newState(): EntityEditorState {
  return initializeEntityEditorState({ schema });
}

function stateWithoutSchema(state: EntityEditorState) {
  const { schema, ...newState } = state;
  newState.drafts = state.drafts.map((draftState) => {
    const newDraftState = { ...draftState };
    if (newDraftState.entity) {
      const { entitySpec, ...newEntity } = newDraftState.entity;
      newEntity.fields = newEntity.fields.map((field) => {
        const { fieldSpec, ...newField } = field;
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        return newField as any;
      });
      // eslint-disable-next-line @typescript-eslint/no-explicit-any
      newDraftState.entity = newEntity as any;
    }
    return newDraftState;
  });
  return newState;
}

describe('reduceEntityEditorState', () => {
  test('AddDraftAction new entity', async () => {
    const id = 'e4e78fce-1089-41b8-9b6b-440d2e044061';
    let state = newState();
    state = reduceEntityEditorState(state, new AddEntityDraftAction({ id, newType: 'Foo' }));
    expect(stateWithoutSchema(state)).toMatchSnapshot();
  });

  test('SetNameAction new entity', () => {
    const id = insecureTestUuidv4();
    let state = newState();
    state = reduceEntityEditorState(state, new AddEntityDraftAction({ id, newType: 'Foo' }));
    state = reduceEntityEditorState(state, new SetNameAction(id, 'New name'));
    expect(state.drafts[0].entity?.name).toEqual('New name');
  });

  test('UpdateEntityAction', async () => {
    let state = newState();
    state = reduceEntityEditorState(state, new AddEntityDraftAction({ id: bar1Id }));
    state = reduceEntityEditorState(
      state,
      new UpdateEntityAction(bar1Id, {
        id: bar1Id,
        info: {
          type: 'Bar',
          authKey: 'none',
          createdAt: Temporal.Instant.from('2021-08-17T07:51:25.56Z'),
          updatedAt: Temporal.Instant.from('2021-08-17T07:51:25.56Z'),
          name: 'Hello',
          version: 0,
          status: AdminEntityStatus.draft,
        },
        fields: { title: 'Hello' },
      })
    );
    expect(stateWithoutSchema(state)).toMatchSnapshot();
  });
});
