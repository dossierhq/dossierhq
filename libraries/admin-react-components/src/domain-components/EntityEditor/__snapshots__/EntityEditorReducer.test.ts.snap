// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reduceEntityEditorState AddDraftAction new entity 1`] = `
Object {
  "activeEntityId": "e4e78fce-1089-41b8-9b6b-440d2e044061",
  "drafts": Array [
    Object {
      "entity": Object {
        "authKey": null,
        "fields": Array [
          Object {
            "initialValue": null,
            "value": null,
          },
          Object {
            "initialValue": null,
            "value": null,
          },
          Object {
            "initialValue": null,
            "value": null,
          },
          Object {
            "initialValue": null,
            "value": null,
          },
          Object {
            "initialValue": null,
            "value": null,
          },
          Object {
            "initialValue": null,
            "value": null,
          },
          Object {
            "initialValue": null,
            "value": null,
          },
          Object {
            "initialValue": null,
            "value": null,
          },
          Object {
            "initialValue": null,
            "value": null,
          },
          Object {
            "initialValue": null,
            "value": null,
          },
          Object {
            "initialValue": null,
            "value": null,
          },
        ],
        "initialName": "",
        "name": "",
        "version": 0,
      },
      "entityLoadMessage": null,
      "exists": false,
      "id": "e4e78fce-1089-41b8-9b6b-440d2e044061",
      "initMessage": null,
      "latestServerVersion": null,
      "status": null,
    },
  ],
}
`;

exports[`reduceEntityEditorState UpdateEntityAction 1`] = `
Object {
  "activeEntityId": "cb228716-d3dd-444f-9a77-80443d436339",
  "drafts": Array [
    Object {
      "entity": Object {
        "authKey": "none",
        "fields": Array [
          Object {
            "initialValue": "Hello",
            "value": "Hello",
          },
        ],
        "initialName": "Hello",
        "name": "Hello",
        "version": 1,
      },
      "entityLoadMessage": null,
      "exists": true,
      "id": "cb228716-d3dd-444f-9a77-80443d436339",
      "initMessage": null,
      "latestServerVersion": 0,
      "status": "draft",
    },
  ],
}
`;
